packages/cmrjs/src/cmr-utils.js

```diff
@@ -140,22 +146,32 @@ async function publishECHO10XML2CMR(cmrFile, cmrClient) {
  * @param {Object} cmrFile.metadataObject - the UMMG JSON cmr metadata
  * @param {Object} cmrFile.granuleId - the metadata's granuleId
  * @param {Object} cmrClient - a CMR instance
+ * @param {string} revisionId - Optional CMR Revision ID
  * @returns {Object} CMR's success response which includes the concept-id
  */
-async function publishUMMGJSON2CMR(cmrFile, cmrClient) {
+async function publishUMMGJSON2CMR(cmrFile, cmrClient, revisionId = undefined) {
   const granuleId = cmrFile.metadataObject.GranuleUR;
-
   const res = await cmrClient.ingestUMMGranule(cmrFile.metadataObject);
+
   const conceptId = res['concept-id'];
+  const resultingRevisionId = res['revision-id'];
+
+  const filename = getS3UrlOfFile(cmrFile);
+  const metadataFormat = ummVersionToMetadataFormat(ummVersion(cmrFile.metadataObject));
+  const link = constructCmrConceptLink(conceptId, 'umm_json');
+
+  if (revisionId && (resultingRevisionId !== revisionId)) {
```

Comment:
I'd have to ask @laurenfrederick how this actually works, but my understanding (that may be wrong) from the ticket was that CMR itself throws an error if you try to post to CMR using an outdated revision ID

Reply:
I do think that is true. We could verify in the #cmr channel.

Reply:
@botanical Can you investigate this by asking in the #cmr channel? I don't think we need to throw the error from our code - I think CMR will throw an error if you send a request using an outdated revision ID

Reply:
I confirmed in the CMR channel